services/datamanager/builtin/builtin.go

```diff
@@ -425,6 +439,13 @@ func (svc *builtIn) Reconfigure(
 		svc.collectors = make(map[resourceMethodMetadata]*collectorAndConfig)
 	}
 
+	if svc.fileDeletionRoutineCancelFn != nil {
+		svc.fileDeletionRoutineCancelFn()
+	}
+	if svc.fileDeletionWg != nil {
+		svc.fileDeletionWg.Wait()
```

Comment:
How long does this take locally? As discussed, we'll want benchmark this when the worker is carrying out real deletion logic, since we don't want reconfigure to time out.

Reply:
Right now the wait group is terminated as soon as the context is. This will remain true in the future. My plan is once I implement real file deletion, I'll check in each iteration if the context is cancelled so if should return quickly